src/cmd-buildextend-live

```diff
@@ -91,42 +95,69 @@ name_version = f'{base_name}-{args.build}'
 # to shorten this more intelligently, otherwise we truncate the
 # version which may impede uniqueness.
 volid = name_version[0:32]
+build_path = Path(f"./{builddir}/{base_name}-{args.build}").resolve()
+kernel_name = f'{base_name}-{args.build}-live-kernel-{basearch}'
+initramfs_name = f'{base_name}-{args.build}-live-initramfs.{basearch}.img'
+rootfs_name = f'{base_name}-{args.build}-live-rootfs.{basearch}.img'
+kernel_file = os.path.join(builddir, kernel_name)
+initramfs_file = os.path.join(builddir, initramfs_name)
+rootfs_file = os.path.join(builddir, rootfs_name)
+
+data = {
+    "buildid": args.build,
+    "imgid": iso_name,
+    "ostree-commit": buildmeta_commit,
+    "container-imgref": "",
+    "deploy-via-container": "",
+    "osname": base_name,
+    "ostree-ref": args.build,
+    "ostree-container": f"{build_path}-ostree.{basearch}.ociarchive",
+    "metal-filename": f"{build_path}-metal.{basearch}.raw",
+    "metal4k-filename": f"{build_path}-metal4k.{basearch}.raw",
+    "ostree-repo": repo,
+    "extra-kargs-string": "mitigations=auto,nosmt",
+    "image-type": "live-iso",
+    "cloud-image-size": "10240",
+    "metal-image-size": "2405",
+    "squashfs-compression": squashfs_compression,
+    "rootfs-size": 0,
+    "live-efiboot-img-size": 16
+}
+
+image_for_disk_json = "runvm.json"
+with open(image_for_disk_json, 'w') as file:
+    json.dump(data, file, indent=4)
+
+
+def update_buildmeta():
+    buildmeta['images'].update({
+        'live-iso': {
+            'path': iso_name,
+            'sha256': sha256sum_file(tmpisofile),
+            'skip-compression': True,
+        }
+    })
+    buildmeta['images'].update({
+        'live-kernel': {
+            'path': kernel_name,
+            'sha256': sha256sum_file(kernel_file),
+            'skip-compression': True,
+        },
+        'live-initramfs': {
+            'path': initramfs_name,
+            'sha256': sha256sum_file(initramfs_file),
+            'skip-compression': True,
+        },
+        'live-rootfs': {
+            'path': rootfs_name,
+            'sha256': sha256sum_file(rootfs_file),
+            'skip-compression': True,
+        }
+    })
 
-tmpdir = os.environ.get("FORCE_TMPDIR", f"{workdir}/tmp/buildpost-live")
-if os.path.isdir(tmpdir):
-    shutil.rmtree(tmpdir)
-
-tmpisoroot = os.path.join(tmpdir, 'live')
-tmpisocoreos = os.path.join(tmpisoroot, 'coreos')
-tmpisoimages = os.path.join(tmpisoroot, 'images')
-tmpisoimagespxe = os.path.join(tmpisoimages, 'pxeboot')
-tmpisoisolinux = os.path.join(tmpisoroot, 'isolinux')
-# contents of initramfs on both PXE and ISO
-tmpinitrd_base = os.path.join(tmpdir, 'initrd')
-# contents of rootfs image
-tmpinitrd_rootfs = os.path.join(tmpdir, 'initrd-rootfs')
-
-for d in (tmpdir, tmpisoroot, tmpisocoreos, tmpisoimages, tmpisoimagespxe,
-        tmpisoisolinux, tmpinitrd_base, tmpinitrd_rootfs):
-    os.mkdir(d)
-
```

Comment:
Also would be good to have this be a separate prep commit that puts it in `generate_iso()`.